fi/include/reverse_purge_hash_map.hpp

```diff
@@ -107,8 +113,8 @@ class reverse_purge_hash_map<K, V, H, E, A>::iterator: public std::iterator<std:
   iterator operator++(int) { iterator tmp(*this); operator++(); return tmp; }
   bool operator==(const iterator& rhs) const { return count == rhs.count; }
   bool operator!=(const iterator& rhs) const { return count != rhs.count; }
-  const std::pair<K&, V> operator*() const {
-    return std::pair<K&, V>(map->keys_[index], map->values_[index]);
+  const value_type operator*() const {
+    return value_type(map->keys_[index], map->values_[index]);
```

Comment:
and instantiating value_type here is fine as implementation, I suppose

Reply:
I don't see how the implementation can do anything else? It seems weird to implement as `reference(key, value)`